core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -387,9 +387,15 @@ public boolean sync() throws AccessControlException, InvalidPathException {
           syncPathCount + failedSyncPathCount, syncPathCount, failedSyncPathCount, end - start,
           mRootScheme);
     }
+    boolean success = syncPathCount > 0;
```

Comment:
slightly concerned about failure and cancellations, because we would mark the directory as synced when perhaps only one file in the directory is synced. It would also prevent it from everything in the directory getting synced for a while..

Reply:
Yeah, but I think that is true even with the old code, since once the directory inode itself is synced, it is marked as synced, even before any of its children are even processed.

Reply:
can we declare the sync successful if the failedCount == 0 and we are not canceled (nothing left in the queue?)
Would that be a better condition for synced?

Reply:
Currently the `failedCount` even includes paths which did not require sync, so `failedCount` is not an accurate name.

Reply:
is it possible to split that out? consider a directory containing only a directory that is already synced. Would the `syncPathCount` be 0 and the parent directory would never be considered synced?